stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt

```diff
@@ -346,12 +346,19 @@ internal class ChannelStateLogicImpl(
     /**
      * Deletes a member. Doesn't delete in the database.
      *
-     * @param userId Id of the user.
+     * @param member The member to be removed.
      */
-    override fun deleteMember(userId: String) {
-        mutableState.rawMembers = mutableState.rawMembers - userId
+    override fun deleteMember(member: Member) {
+        val user = member.user
+
+        mutableState.rawMembers = mutableState.rawMembers - user.id
 
         mutableState.setMembersCount(mutableState.membersCount.value - 1)
+
+        if (user.online) {
+            deleteWatcher(user)
+            setWatcherCount(mutableState.watcherCount.value - 1)
+        }
```

Comment:
Same here, If we already have a list of watcher, we shouldn't need to have a separate counter for something our collection already know.
On the other hand, as far as we are using `StateFlow`, this property (The counter) could be a "map" of the other property (The collection) and it will be always up-to-date

Reply:
I think the list of watchers is also paginated. There can be 1000 watchers for a live stream channel, but I don't think all the watchers will be returned for the channel.